urllib3/response.py

```diff
@@ -327,9 +389,19 @@ def read(self, amt=None, decode_content=None, cache_content=False):
                     # no harm in redundantly calling close.
                     self._fp.close()
                     flush_decoder = True
+                    if (self.enforce_content_length and
+                            self.length_remaining not in (0, None)):
```

Comment:
Cool, it's a length of 95 expanded, so I went with that approach. Adding the check as another method seems a bit excessive if we're only going to use it here.